src/vs/workbench/browser/parts/editor/tabsTitleControl.ts

```diff
@@ -337,8 +339,25 @@ export class TabsTitleControl extends TitleControl {
 				}
 			}
 
-			// Figure out scrolling direction
-			const nextEditor = this.group.getEditorByIndex(this.group.getIndexOfEditor(activeEditor) + (e.deltaX < 0 || e.deltaY < 0 /* scrolling up */ ? -1 : 1));
+			// Ignore event if the last one happened too recently.
+			// The restriction is relaxed according to the abs value of deltaX and deltaY
+			const time_threshold = 150;
+			if (Date.now() - this.lastMouseWheelEventTime < time_threshold - 2 * (Math.abs(e.deltaX) + Math.abs(e.deltaY))) {
+				return;
+			}
+			this.lastMouseWheelEventTime = Date.now();
+
+			// Figure out scrolling direction (ignore direction if the scrolling is too subtile)
+			const distance_threshold = 1.5;
+			let tabSwitchDirection = 0;
+			if (e.deltaX + e.deltaY < - distance_threshold) {
+				tabSwitchDirection = -1;
+			}
+			if (e.deltaX + e.deltaY > distance_threshold) {
+				tabSwitchDirection = 1;
+			}
```

Comment:
```suggestion
			if (e.deltaX + e.deltaY < - distance_threshold) {
				tabSwitchDirection = -1;
			} else if (e.deltaX + e.deltaY > distance_threshold) {
				tabSwitchDirection = 1;
			}
```

Reply:
You're right. For some reason I initially thought both conditions could be true, but that's obviously not the case.